src/monotonic/monotonic.jl

```diff
@@ -180,7 +180,12 @@ function (itp::MonotonicInterpolation)(x::Number)
     return itp.A[k] + itp.m[k]*xdiff + itp.c[k]*xdiff*xdiff + itp.d[k]*xdiff*xdiff*xdiff
 end
 
-function derivative(itp::MonotonicInterpolation, x::Number)
+function gradient(itp::MonotonicInterpolation, x::AbstractArray{<:Number, 1})
+    length(x) == 1 || error("Given vector x ($x) should have exactly one element")
```

Comment:
With `x::VarArg{<:Number,1}`, I don't think this check is needed (but I'm not sure).

Reply:
This check isn't needed with `function gradient(itp::MonotonicInterpolation, x::Number)` and it works.

Reply:
Yeah; for monotonic interpolations it doesn't matter very much.

My main concern here is that I want the interface to match `gradient` for other interpolation types _exactly_, with the only exception being the type of `itp`. The idea is that you should be able to have code that initializes an interpolation object of some type, and then (perhaps a lot later in the code) use any part of the common API without caring which type you used. So if you e.g. start out with regular B-splines, and then realize you want monotonic splines, you switch out the initialization part _and nothing else_.

It might be that `x::Number` accomplishes that well enough, and if so it's totally fine by me. I just want to make sure that the signature of this method (as well as the others, e.g. `itp(<whatever>)` and `hessian(itp, <whatever>)`) is designed with the common API in mind.

Reply:
Yes, I understand, but `Vararg{<:Number,1}` didn't work (I wrote about it here: https://github.com/JuliaMath/Interpolations.jl/pull/238#discussion_r221170008 ). `gradient(itp, [1.0])` still works (`UnexpandedIndexTypes` union contains `AbstractVector`). Is there another use case I should keep in mind?

Reply:
Hum. I'll ping in @timholy here...